src/genie/libs/parser/iosxe/show_ap.py

```diff
@@ -0,0 +1,258 @@
+        # ab21-cap44                    5c50.1507.f180  1       Enabled        Up            20     *7/8 (6 dBm)    (132)*
+        #
+        # AP Name                           Mac Address     Slot    Admin State    Oper State    Width  Txpwr           Channel
+        # ---------------------------------------------------------------------------------------------------------------------------------
+        # ab12-cap8                     5c50.1507.f1a0  1       Enabled        Up            20     *8/8 (5 dBm)    (157)*
+        # ab32-cap15                    5c50.1507.f200  1       Enabled        Up            20     *6/8 (9 dBm)    (116)*
+        # ab22-cap13                    5c50.1507.f2c0  1       Enabled        Up            20     *8/8 (3 dBm)    (112)*
+
+        regex_patterns = {
+            'ap_dot11_5ghz_summ': re.compile(r"^(?P<ap_name>\S+)\s+(?P<mac_addres>\w+\.\w+\.\w+)\s+(?P<slot>\d+)\s+(?P<admin_state>(Enabled|Disabled))\s+(?P<oper_state>(Up|Down))\s+(?P<width>\d+)\s+(?P<tx_pwr>.*)\s+(?P<channel>\(\d+\)\*)")
+        }
+
+        remove_lines = ('---', 'AP Name ')
+
+        # Remove unwanted lines from raw text
+        def filter_lines(raw_output, remove_lines):
+            # Remove empty lines
+            clean_lines = list(filter(None, raw_output.splitlines()))
+            rendered_lines = []
+            for clean_line in clean_lines:
+                clean_line_strip = clean_line.strip()
+                # Remove lines unwanted lines from list of "remove_lines"
+                if not clean_line_strip.startswith(remove_lines):
+                    rendered_lines.append(clean_line_strip)
+            return rendered_lines
+
+
+        def match_regex(line, regex_patterns):
+            groups = {}
+            for capture, pattern in regex_patterns.items():
+                if pattern.match(line):
+                    match = pattern.match(line)
+                    groups = match.groupdict()
+            return groups
+
+
+        def change_data_type(value):
+            value = value.strip()
+            if value.isdigit():
+                value = int(value)
+            else:
+                try:
+                    # Change strings to float if possible
+                    value = float(value)
+                except ValueError:
+                    # if the value is not an int or float, leave it as a string.
+                    pass
+            return value
+
+        out_filter = filter_lines(raw_output=out, remove_lines=remove_lines)
+        # Key for each capture group
+        ap_name_key = ''
+
+        for line in out_filter:
+            # ab32-cap15                    5c50.1507.f200  1       Enabled        Up            20     *6/8 (9 dBm)    (116)*
+            captured_groups = match_regex(line=line, regex_patterns=regex_patterns)
+            if captured_groups:
+                # Define the root key
+                if not ap_dot11_5ghz_summ_dict.get('ap_dot11_5ghz_summ', {}):
```

Comment:
Why is this needed? Could it not be `ap_dot11_5ghz_summ_dict['ap_dot11_5ghz_summ'] = {}` before the for loop?

Reply:
I updated and removed the root key. There was a Cisco comment on another PR to remove if it's not technically needed. [Cisco PR#212](https://github.com/CiscoTestAutomation/genieparser/pull/212/files#r477211500)